Examples/Pass/ticketLockNonAtomicRelease2.cvf

```diff
@@ -0,0 +1,55 @@
+/*
+ * Linux-style ticketed lock.
+ */
+
+global int ticket;  // The next ticket to hand out.
+global int serving; // The current ticket holding the lock.
+local int t;  // The thread's current ticket.
+local int s;  // The thread's current view of serving.
+
+/*
+ * Locks the ticket lock.
+ */
+method lock() {
+  {| emp |}
+    <t = ticket++>;
+  {| holdTick(t) |}
+    do {
+      {| holdTick(t) |}
+        <s = serving>;
+      {| if s == t then holdLock() else holdTick(t) |}
+    } while (s != t);
+  {| holdLock() |}
+}
+
+/*
+ * Unlocks the ticket lock.
+ */
+method unlock() {
+  {| holdLock() |}
+    <s = serving>;
+  {| /*relLock(s)*/ holdLock() |}
```

Comment:
I don't see why this proof goes through?  s is not mentioned, so it seems it can take any value, and thus I don't see why the next command is sound.  Thoughts?  Is this a bug?